engines/macventure/gui.cpp

```diff
@@ -371,14 +400,51 @@ void Gui::loadBorders(Graphics::MacWindow *target, MVWindowType type) {
 
 void Gui::loadBorder(Graphics::MacWindow *target, MVWindowType type, bool active) {
 
-	Common::SeekableReadStream *stream = _engine->getBorderFile(type, active);
+    Common::SeekableReadStream *stream;
 
-	if (stream) {
-		BorderBounds bbs = borderBounds(type);
-		target->loadBorder(*stream, active, bbs.leftOffset, bbs.rightOffset, bbs.topOffset, bbs.bottomOffset);
+    bool activeFlag = (active ? Graphics::kWindowBorderActive : 0);    
+    bool canHaveTitle = false;
+    bool canHaveScrollbar = false;
 
-		delete stream;
-	}
+    if (type == MacVenture::kInvWindow || type == MacVenture::kZoomDoc) {
+        canHaveScrollbar = true;
+    }
+
+    if (type == MacVenture::kInvWindow || type == MacVenture::kZoomDoc || type == MacVenture::kNoGrowDoc || type == MacVenture::kRDoc4) {
+        canHaveTitle = true;
+    }
+    
+	Graphics::BorderOffsets offsets = borderOffsets(type);
+    stream = _engine->getBorderFile(type, active);
+    
+	if (stream) {
+        target->loadBorder(*stream, activeFlag, offsets);
+        delete stream;
+    }
+
+    if (canHaveTitle) {
+        stream = _engine->getBorderFile(type, active);
```

Comment:
I wonder why you are allocating a new stream for the same thing each time (at least judging from the diff I can see here)?

It should really be enough to do that once but reset it each time you use is for other stuff. Would also simplify the code because you can early opt-out if the stream is `nullptr`